### PR TITLE
Bump grimoirelab-toolkit from 0.1.12 to 0.2.0

### PR DESCRIPTION
**Pull request by vchrombie**
Bump grimoirelab-toolkit from 0.1.12 to 0.2.0

### CODE REVIEW

**Review by sduenas — APPROVED** [review]
LGTM

**Comment by sduenas** [comment]
@vchrombie is this still in WIP?

**Comment by vchrombie** [comment]
> @vchrombie is this still in WIP?

Hi @sduenas, thanks for reviewing.
I would like you not to merge this PR right now.

I'm testing something with ELK, so I will let you know once everything is resolved.

**Comment by vchrombie** [comment]
This PR is an extension to https://github.com/chaoss/grimoirelab-elk/pull/988.
@sduenas @zhquan, it would be great if you can review it when you are free.

File: poetry.lock [hunk 1]
```diff
@@ -434,10 +441,7 @@ flake8 = [
     {file = "flake8-3.9.1-py2.py3-none-any.whl", hash = "sha256:3b9f848952dddccf635be78098ca75010f073bfe14d2c6bda867154bea728d2a"},
     {file = "flake8-3.9.1.tar.gz", hash = "sha256:1aa8990be1e689d96c745c5682b687ea49f2e05a443aff1f8251092b0014e378"},
 ]
-grimoirelab-toolkit = [
-    {file = "grimoirelab-toolkit-0.1.12.tar.gz", hash = "sha256:de071c29349e3dfdc0b6631c7664b2d8374c7e6c46f994725f6617ff5eebec78"},
-    {file = "grimoirelab_toolkit-0.1.12-py3-none-any.whl", hash = "sha256:b77db5f77fe162b9fef2db3f721f994378bfa37485ce07ebb7a40e762e7899e1"},
-]
+grimoirelab-toolkit = []
```
Comment (sduenas) [inline poetry.lock:444]: Not sure about why this?. Any idea?

Comment (vchrombie) [reply poetry.lock:444]: I don't know about it, I have used the poetry commands and it gives this result.
```
p0tt3r@wand ~/dev/sources/grimoirelab-perceval              
(.venv) >🔮$ poetry add grimoirelab_toolkit@^0.2.0

Updating dependencies
Resolving dependencies... (0.3s)

Writing lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating grimoirelab-toolkit (0.1.12 -> 0.2.0)
                                                            
p0tt3r@wand ~/dev/sources/grimoirelab-perceval              
(.venv) >🔮$ poetry add -D git+https://github.com/chaoss/grimoirelab-toolkit.git@master

Updating dependencies
Resolving dependencies... (1.6s)

Writing lock file

No dependencies to install or update
```

Comment (sduenas) [reply poetry.lock:444]: Seems a bit strange for me. How does it know what to do when it needs to install the pip package and not the dev dependencies? What does it do?

Btw, I think it should be `poetry add grimoirelab_toolkit@^0.2` and not `0.2.0`.

Comment (vchrombie) [reply poetry.lock:444]: > Seems a bit strange for me. How does it know what to do when it needs to install the pip package and not the dev dependencies? What does it do?

Let me try to install using `--no-dev`.

> Btw, I think it should be poetry add grimoirelab_toolkit@^0.2 and not 0.2.0.

Okay, will update the PR now.

Comment (vchrombie) [reply poetry.lock:444]: Surprising :open_mouth: 
```
p0tt3r@wand ~/dev/sources/grimoirelab-perceval              
>🔮$ poetry install --no-dev
Creating virtualenv perceval in /home/p0tt3r/dev/sources/grimoirelab-perceval/.venv
Installing dependencies from lock file

Package operations: 16 installs, 0 updates, 0 removals

  • Installing pycparser (2.20)
  • Installing six (1.15.0)
  • Installing certifi (2020.12.5)
  • Installing cffi (1.14.4)
  • Installing chardet (4.0.0)
  • Installing idna (2.10)
  • Installing python-dateutil (2.8.1)
  • Installing soupsieve (2.1)
  • Installing urllib3 (1.26.3)
  • Installing beautifulsoup4 (4.9.3)
  • Installing cryptography (3.3.1)
  • Installing dulwich (0.20.15)
  • Installing feedparser (5.2.1)
  • Installing grimoirelab-toolkit (0.2.0 30b0253)
  • Installing pyjwt (1.7.1)
  • Installing requests (2.25.1)

Installing the current project: perceval (0.17.4)
```

It seems it is still using git source URL for installing the dependency.

It usually shows the commit latest commit hash if using git source URL while installing
```
  • Installing grimoirelab-toolkit (0.2.0 30b0253)
```
in other cases, it looks like this
```
  • Installing grimoirelab-toolkit (0.2.0)
```

Comment (vchrombie) [reply poetry.lock:444]: Another small observation, I saw what version does it use when poetry builds a package using poetry, `poetry build`.

The `setup.py` consists of the version number.
```
install_requires = \
['PyJWT>=1.7.1,<2.0.0',
 'beautifulsoup4>=4.3.2,<5.0.0',
 'cryptography>=3.3.1,<4.0.0',
 'dulwich>=0.20.0,<0.21.0',
 'feedparser>=5.1.3,<6.0.0',
 'grimoirelab-toolkit>=0.2,<0.3',
 'python-dateutil>=2.6.0,<3.0.0',
 'requests>=2.7.0,<3.0.0',
 'urllib3>=1.22,<2.0']
```

Comment (vchrombie) [reply poetry.lock:444]: Looks like this is an existent problem but in a different version, please refer https://github.com/python-poetry/poetry/issues/2837 and https://github.com/python-poetry/poetry/issues/1168.

> apparent answer is that it _always_ obeys the details of the dev-dependency

Comment (sduenas) [reply poetry.lock:444]: Correct me if I'm wrong but that means we can't so what we were trying to do.

Comment (vchrombie) [reply poetry.lock:444]: Yup, you are right.

**Review by sduenas — APPROVED** [review]
LGTM

**Comment by vchrombie** [comment]
We are updating the `grimoirelab_toolkit` version for now, as the other PRs are blocked by this. We would come back again and think about how to add grimoirelab components as the dependencies.

**Pull request merged**